src/image.rs

```diff
@@ -121,7 +121,11 @@ impl<'docker> Images<'docker> {
         // stream. But for backwards compatability, we have to return the error inside of the
         // stream.
         let mut bytes = Vec::default();
-        let tar_result = tarball::dir(&mut bytes, opts.path.as_str());
+
+        // Check if skip_gzip build option is set to true. If yes, we avoid building a tar object
+        // of the docker image while building it
+        let skip_gzip = opts.params.get("skip_gzip").unwrap_or(&"None".to_string()).eq("true");
```

Comment:
I think `skip_gzip` can be a new field in `BuildOptions` struct and it can be a bool type. This way you won't need to compare it with string, also `params` is used for parameters serialized into the docker daemon and yours is not part of that.

Reply:
Updated the PR to add a new boolean field to BuildOptions